crml/staking/src/rewards/mod.rs

```diff
@@ -151,17 +172,88 @@ where
 				Self::calculate_npos_payouts(&validator, *validator_commission, stake_map, total_payout_share)
 			})
 			.for_each(|(account, payout)| {
-				total_payout_imbalance.maybe_subsume(T::CurrencyToReward::deposit_into_existing(&account, payout).ok());
+				Payouts::<T>::mutate(|p| p.push_back((account, payout, era)));
 			});
 
-		// Any unallocated reward amount can go to the development fund
-		let remainder = total_payout.saturating_sub(total_payout_imbalance.peek());
-		total_payout_imbalance.subsume(
-			T::CurrencyToReward::deposit_into_existing(&T::TreasuryModuleId::get().into_account(), remainder)
-				.unwrap_or_else(|_| PositiveImbalanceOf::<T>::zero()),
-		);
+		RemainingRewardAmounts::<T>::mutate(|rra| {
+			rra.push_back(total_payout.saturating_sub(total_payout_imbalance.peek()))
+		});
```

Comment:
Can use the already calculated amount
could also unit-test that `RemainingRewardAmounts` is being set correctly by this method
```suggestion
		RemainingRewardAmounts::<T>::mutate(|rra| {
			rra.push_back(validator_payout)
		});
```

Reply:
Addressed by https://github.com/cennznet/cennznet/pull/338/commits/b7844ebfb512b3525f12d67f193dd3fbb11a7742